src/client/datascience/kernel-launcher/localKernelFinder.ts

```diff
@@ -147,33 +105,27 @@ export class LocalKernelFinder implements ILocalKernelFinder {
         cancelToken?: CancellationToken
     ): Promise<LocalKernelConnectionMetadata[]> {
         try {
-            // Get an id for the workspace folder, if we don't have one, use the fsPath of the resource
-            const workspaceFolderId =
-                this.workspaceService.getWorkspaceFolderIdentifier(
-                    resource,
-                    resource?.fsPath || this.workspaceService.rootPath
-                ) || 'root';
-
-            // If we have not already searched for this resource, then generate the search
-            if (workspaceFolderId && !this.workspaceToMetadata.has(workspaceFolderId)) {
-                this.workspaceToMetadata.set(
-                    workspaceFolderId,
-                    this.findResourceKernelMetadata(resource, cancelToken).then((items) => {
-                        traceInfoIf(
-                            !!process.env.VSC_JUPYTER_LOG_KERNEL_OUTPUT,
-                            `Kernel specs for ${resource?.toString() || 'undefined'} are \n ${JSON.stringify(
-                                items,
-                                undefined,
-                                4
-                            )}`
-                        );
-                        return items;
-                    })
-                );
-            }
+            const [nonPythonKernelSpecs, pythonRelatedKernelSpecs] = await Promise.all([
+                this.nonPythonkernelFinder.listKernels(resource, cancelToken),
+                this.extensionChecker.isPythonExtensionInstalled
+                    ? this.pythonKernelFinder.listKernels(resource, cancelToken)
+                    : []
+            ]);
+
+            const kernels = [...nonPythonKernelSpecs, ...pythonRelatedKernelSpecs].filter(({ kernelSpec }) => {
```

Comment:
The results from Python & Non Python are merged here.
Hence we should still get the same list of kernels.